docs/manual/select_and_filter_data.rst

```diff
@@ -0,0 +1,1173 @@
+      |  floats
+      | float64
+   -- + -------
+    0 |    10  
+    1 |    11.5
+    2 |    12.3
+    3 |   -13  
+   [4 rows x 1 column]
+
+   >>> DT[:, dt.ltype.time]
+      | dates     
+      | date32    
+   -- + ----------
+    0 | 2000-01-05
+    1 | 2010-11-23
+    2 | 2020-02-29
+    3 | NA             
+   [4 rows x 1 column]
+
+A list of types can be selected as well::
+
+   >>> DT[:, [date, str]]
+      | dates       strings
+      | date32      str32  
+   -- + ----------  -------
+    0 | 2000-01-05  A      
+    1 | 2010-11-23  B      
+    2 | 2020-02-29  NA     
+    3 | NA          D         
+   [4 rows x 2 columns]
+
+By List/Tuple
+^^^^^^^^^^^^^
+
+Using a sequence allows for selection of multiple columns::
+
+   >>> DT[:, ['integers', 'strings']]
+      | integers  strings
+      |    int32  str32  
+   -- + --------  -------
+    0 |        1  A      
+    1 |        2  B      
+    2 |        3  NA     
+    3 |        4  D      
+   [4 rows x 2 columns]
+   
+   >>> DT[:, (-3, 2, 3)] 
+      | integers   floats  strings
+      |    int32  float64  str32  
+   -- + --------  -------  -------
+    0 |        1     10    A      
+    1 |        2     11.5  B      
+    2 |        3     12.3  NA     
+    3 |        4    -13    D      
+   [4 rows x 3 columns]
+
+Selection via `list comprehension`_/`generator expression`_ is possible::
+
+   >>> DT[:, [num for num in range(DT.ncols) if num % 2 == 0]]
+      | dates        floats
+      | date32      float64
+   -- + ----------  -------
+    0 | 2000-01-05     10  
+    1 | 2010-11-23     11.5
+    2 | 2020-02-29     12.3
+    3 | NA            -13  
+   [4 rows x 2 columns]
+
+
+.. note:: Selecting columns via a mix of column names and position(integers) is not allowed in datatable.
+
+   >>> DT[:, ['dates', 2]]
+   >>> ---------------------------------------------------------------------------
+   >>> TypeError                                 Traceback (most recent call last)
+   >>> <ipython-input-82-af15fff98d55> in <module>
+   >>> ----> 1 DT[:, ['dates', 2]]
+   >>>
+   >>> TypeError: Mixed selector types are not allowed. 
+   >>>            Element 1 is of type integer, 
+   >>>            whereas the previous element(s) were of type string
+
+Via slicing
+^^^^^^^^^^^
+When slicing on strings, both the ``start`` and ``end`` column names are included in the returned frame
+::
+
+   >>> DT[:, 'dates' : 'strings']
+      | dates       integers   floats  strings
+      | date32         int32  float64  str32  
+   -- + ----------  --------  -------  -------
+    0 | 2000-01-05         1     10    A      
+    1 | 2010-11-23         2     11.5  B      
+    2 | 2020-02-29         3     12.3  NA     
+    3 | NA                 4    -13    D      
+   [4 rows x 4 columns]
+
+However, when slicing via position, the columns are returned up to, but not including the final position; this is similar to the slicing pattern for Python's sequences
+
+   >>> DT[:, 1 : 3] 
```

Comment:
May be format slices as `x:y` (without spaces) then it would be easier to distinguish i and j selectors? Compare `DT[:, 1 : 3]` and `DT[:, 1:3]`.